nautobot/ipam/migrations/0037_vlan_vlan_group_default_values.py

```diff
@@ -0,0 +1,52 @@
+# Generated by Django 3.2.18 on 2023-06-07 16:31
+
+from django.db import migrations
+
+from nautobot.ipam.utils.migrations import increment_names_of_records_with_similar_names
+
+
+DEFAULT_VLAN_GROUP_BASENAME = "Default VLAN Group"
+
+
+def vlan_vlan_group_uniqueness_constrains_revise(apps, schema_editor):
+    # Resolve VLAN-Group name uniqueness
+    VLAN = apps.get_model("ipam", "vlan")
+    VLANGroup = apps.get_model("ipam", "vlangroup")
+
+    increment_names_of_records_with_similar_names(VLANGroup)
+
+    vlans_without_vlan_groups = VLAN.objects.filter(vlan_group__isnull=True)
+    vlans_to_update = []
+    counter = 1
+    for vlan in vlans_without_vlan_groups:
+        while True:
+            created, vlan_group = VLANGroup.objects.get_or_create(
+                name=f"{DEFAULT_VLAN_GROUP_BASENAME} {counter}", defaults={"location": vlan.location}
+            )
+            if created or vlan_group.location == vlan.location:
+                vlan.vlan_group = vlan_group
+                vlans_to_update.append(vlan)
+                continue
+            counter += 1
+    VLAN.objects.bulk_update(vlans_to_update, ["vlan_group"])
+
+
+def revert_vlan_vlan_group_uniqueness_constrains_revise(apps, schema_editor):
+    VLAN = apps.get_model("ipam", "vlan")
+    VLANGroup = apps.get_model("ipam", "vlangroup")
+
+    # TODO(timizuo): Find a way to revert changes made to VLAN and VLANGroup
```

Comment:
Any ideas on how I may revert this, since I don't want to destroy a VLAN group that was there before the migration with similar names(highly unlikey),
or should it be none reversible. Same for VLANGroup.

😔

Reply:
I honestly do not think there is a need to reverse this migration. Since we are only adding data in the forward migration. Rolling it back forcefully might cause data loss as you said. but Just my opinion tho.

Reply:
Probably non-reversible is fine.